CI: integration tests - TestSysctl

## CI failure

Hit on https://github.com/cilium/cilium/pull/31781
Link: https://github.com/cilium/cilium/actions/runs/8558031435/job/23451695738

```
--- FAIL: TestSysctl (0.00s)

time="2024-04-04T16:33:40Z" level=info msg=Invoked duration="296.695µs" function="reconciler.Register[...] (pkg/statedb/reconciler/reconciler.go:24)" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg=Invoked duration="68.498µs" function="sysctl.TestSysctl.func2 (pkg/datapath/linux/sysctl/sysctl_test.go:164)" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg=Invoked duration="31.539µs" function="sysctl.TestSysctl.func3 (pkg/datapath/linux/sysctl/sysctl_test.go:167)" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg=Starting subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Start hook executed" duration="1.172µs" function="*statedb.DB.Start" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Start hook executed" duration="9.959µs" function="hive.New.func1.2 (pkg/hive/hive.go:104)" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Start hook executed" duration="1.071µs" function="*cell.reporterHooks.Start" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Start hook executed" duration="4.398µs" function="*job.group.Start" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Start hook executed" duration="1.222µs" function="*job.group.Start" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg=Started duration="76.572µs" subsys=hive
    sysctl_test.go:230: 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/datapath/linux/sysctl/sysctl_test.go:230
        	Error:      	Not equal: 
        	            	expected: "2"
        	            	actual  : "1"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-2
        	            	+1
        	Test:       	TestSysctl
        	Messages:   	unexpected value for parameter {"net.ipv6.conf.all.forwarding" "2" %!q(bool=false) "" " (delete: false, updated: 2562047h47m16.854775807s ago)"}
time="2024-04-04T16:33:40Z" level=info msg=Stopping subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Stop hook executed" duration="12.403µs" function="*job.group.Stop" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Stop hook executed" duration="5.159µs" function="*job.group.Stop" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Stop hook executed" duration="12.764µs" function="*cell.reporterHooks.Stop" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Stop hook executed" duration=761ns function="hive.New.func1.3 (pkg/hive/hive.go:111)" subsys=hive
time="2024-04-04T16:33:40Z" level=info msg="Stop hook executed" duration="7.434µs" function="*statedb.DB.Stop" subsys=hive
```

I managed to reproduce it pretty easily locally with:

```
go test ./pkg/datapath/linux/sysctl/ --count 10000 -run 'TestSysctl$' --test.failfast
```

The actual failure can occur at different stages. Adding a bit of tracing logs to `ops.Update`, this seems a real bug rather than a test flake (note that the last write operation which shouldn't have occurred based on the test definition):

```
...
msg="Read sysctl /proc/sys/net/ipv6/conf/all/forwarding: 0 - expected: 1" subsys=sysctl-test sysParamName=net.ipv6.conf.all.forwarding sysParamValue=1
msg="Writing sysctl /proc/sys/net/ipv6/conf/all/forwarding to 1 (read: 0)" subsys=sysctl-test sysParamName=net.ipv6.conf.all.forwarding sysParamValue=1
msg="Read sysctl /proc/sys/net/ipv4/ip_forward: 1 - expected: 2" subsys=sysctl-test sysParamName=net.ipv4.ip_forward sysParamValue=2
msg="Writing sysctl /proc/sys/net/ipv4/ip_forward to 2 (read: 1)" subsys=sysctl-test sysParamName=net.ipv4.ip_forward sysParamValue=2
msg="Read sysctl /proc/sys/net/ipv4/conf/all/forwarding: 1 - expected: 2" subsys=sysctl-test sysParamName=net.ipv4.conf.all.forwarding sysParamValue=2
msg="Writing sysctl /proc/sys/net/ipv4/conf/all/forwarding to 2 (read: 1)" subsys=sysctl-test sysParamName=net.ipv4.conf.all.forwarding sysParamValue=2
msg="Read sysctl /proc/sys/net/ipv6/conf/all/forwarding: 1 - expected: 2" subsys=sysctl-test sysParamName=net.ipv6.conf.all.forwarding sysParamValue=2
msg="Writing sysctl /proc/sys/net/ipv6/conf/all/forwarding to 2 (read: 1)" subsys=sysctl-test sysParamName=net.ipv6.conf.all.forwarding sysParamValue=2
msg="Read sysctl /proc/sys/net/ipv4/conf/all/forwarding: 2 - expected: 1" subsys=sysctl-test sysParamName=net.ipv4.conf.all.forwarding sysParamValue=1
msg="Writing sysctl /proc/sys/net/ipv4/conf/all/forwarding to 1 (read: 2)" subsys=sysctl-test sysParamName=net.ipv4.conf.all.forwarding sysParamValue=1
```

/cc @pippolo84 @dylandreimerink

--- a/pkg/datapath/linux/sysctl/sysctl_test.go
+++ b/pkg/datapath/linux/sysctl/sysctl_test.go
@@ -158,7 +158,6 @@ func TestSysctl(t *testing.T) {
 				newReconcilerConfig,
 				newOps,
 			),
-			cell.Invoke(reconciler.Register[*tables.Sysctl]),
 		),
 
 		cell.Invoke(func(s Sysctl) {
@@ -262,7 +261,6 @@ func TestSysctlIgnoreErr(t *testing.T) {
 				newReconcilerConfig,
 				newOps,
 			),
-			cell.Invoke(reconciler.Register[*tables.Sysctl]),
 		),
 
 		cell.Invoke(func(s Sysctl) {

--- a/pkg/datapath/linux/sysctl/sysctl_test.go
+++ b/pkg/datapath/linux/sysctl/sysctl_test.go
@@ -227,7 +227,7 @@ func TestSysctl(t *testing.T) {
 	for _, s := range batch {
 		val, err := sysctl.Read(s.Name)
 		assert.NoError(t, err)
-		assert.Equal(t, s.Val, val, "unexpected value for parameter %q", s)
+		assert.Equal(t, s.Val, val, "unexpected value %q for parameter %q", val, s.Name)
 	}
 
 	assert.NoError(t, hive.Stop(context.Background()))

